src/hwcaps.c

```diff
@@ -152,6 +158,17 @@ HardwareCapabilities GetHardwareCapabilities(void) {
   return capabilities;
 }
 
+PlatformType kEmptyPlatformType;
+
+PlatformType GetPlatformType(void) {
+  PlatformType type = kEmptyPlatformType;
+  char *platform = (char *)GetHardwareCapabilitiesFor(AT_PLATFORM);
+  char *base_platform = (char *)GetHardwareCapabilitiesFor(AT_BASE_PLATFORM);
+
+  if (platform != NULL) strcpy(type.platform, platform);
```

Comment:
You need to check that `platform` fits into `type.platform`, if not copy up to `sizeof(type.platform) - 1` and zero terminate the string.

I'll add function to do this in a separate PR.

Reply:
Actually you can use the following snippet to get the job done:
```
if (platform != NULL)
  CpuFeatures_StringView_CopyString(str(platform), type.platform, sizeof(type.platform));
```